apps/aecore/src/aec_block_genesis.erl

```diff
@@ -24,47 +24,46 @@
 
 %% API
 -export([genesis_header/0,
-         genesis_block_as_deserialized_from_network/0,
          genesis_block/0]).
 
 -include("common.hrl").
 -include("blocks.hrl").
 
 genesis_header() ->
-    #header{
-       version = ?GENESIS_VERSION,
-       height = ?GENESIS_HEIGHT,
-       prev_hash = <<0:?BLOCK_HEADER_HASH_BYTES/unit:8>>,
-       txs_hash = <<0:?TXS_HASH_BYTES/unit:8>>,
-       root_hash = <<0:?STATE_HASH_BYTES/unit:8>>,
-       target = ?HIGHEST_TARGET_SCI,
-       pow_evidence = no_value,
-       nonce = 0,
-       time = 0 %% Epoch.
-      }.
-
-genesis_block_as_deserialized_from_network() ->
-    H = genesis_header(),
-    #block{
-       version = H#header.version,
-       height = H#header.height,
-       prev_hash = H#header.prev_hash,
-       txs_hash = H#header.txs_hash,
-       root_hash = H#header.root_hash,
-       target = H#header.target,
-       pow_evidence = H#header.pow_evidence,
-       nonce = H#header.nonce,
-       time = H#header.time,
-       txs = [],
-       trees = _DummyTrees = #trees{}
-      }.
+    aec_blocks:to_header(genesis_block()).
 
 %% Returns the genesis block including the state trees.
 %%
 %% The current implementation of state trees causes a new Erlang term,
-%% representing the initial empty state trees, to be allocated in the
+%% representing the initial state trees, to be allocated in the
 %% heap memory of the calling process.
 genesis_block() ->
-    B = genesis_block_as_deserialized_from_network(),
-    {ok, T} = aec_trees:all_trees_new(),
-    B#block{trees = T}.
+    Trees = populated_trees(),
+    #block{
+      version = ?GENESIS_VERSION,
+      height = ?GENESIS_HEIGHT,
+      prev_hash = <<0:?BLOCK_HEADER_HASH_BYTES/unit:8>>,
+      txs_hash = <<0:?TXS_HASH_BYTES/unit:8>>,
+      root_hash = aec_trees:all_trees_hash(Trees),
+      target = ?HIGHEST_TARGET_SCI,
+      pow_evidence = no_value,
+      nonce = 0,
+      time = 0, %% Epoch.
+      trees = Trees
+      }.
+
+populated_trees() ->
+    {ok, T0} = aec_trees:all_trees_new(),
+    PresetAccounts = aec_genesis_block_settings:preset_accounts(),
```

Comment:
The Merkle tree library depends on the order of operations applied on it. I.e. is same accounts are put in distinct order, a distinct root hash may be obtained. Please comment on this function returning a predictable order of the accounts.